runtime/src/main/java/org/corfudb/runtime/view/AddressSpaceView.java

```diff
@@ -115,14 +118,24 @@ public AddressSpaceView(@Nonnull final CorfuRuntime runtime) {
                 .build();
 
         Optional<MeterRegistry> metricsRegistry = MeterRegistryProvider.getInstance();
-        MicroMeterUtils.gauge("address_space.read_cache.hit_ratio", readCache, cache -> cache.stats().hitRate());
         metricsRegistry.map(registry -> GuavaCacheMetrics.monitor(registry, readCache, "address_space.read_cache"));
+        MicroMeterUtils.gauge("address_space.read_cache.hit_ratio", readCache, cache -> cache.stats().hitRate());
+        MicroMeterUtils.gauge("address_space.read_cache.size", readCache, cache -> cache.size());
+        MicroMeterUtils.gauge("address_space.read_cache.avg_entry_size", calculateEstimatedAvgEntrySize());
     }
 
     private void handleEviction(RemovalNotification<Long, ILogData> notification) {
         if (log.isTraceEnabled()) {
             log.trace("handleEviction: evicting {} cause {}", notification.getKey(), notification.getCause());
         }
+        currentDataSizeInCache.addAndGet(notification.getValue().getSizeEstimate() * -1);
```

Comment:
This can produce a negative average. Image the case where a client just reads data and never writes, or reads at a rate higher than what it writes. It would calculate the wrong thing.